subsys/bluetooth/mesh/main.c

```diff
@@ -59,6 +59,43 @@ int bt_mesh_provision(const u8_t net_key[16], u16_t net_idx,
 		pb_gatt_enabled = false;
 	}
 
+	if (IS_ENABLED(CONFIG_BT_MESH_NETWORK_MGMT)) {
+		struct bt_mesh_mgmt_subnet *subnet;
+		struct bt_mesh_mgmt_node *node;
+		const struct bt_mesh_comp *comp;
+
+		if (!atomic_test_bit(bt_mesh_network.flags,
+				     BT_MESH_DB_VALID)) {
+			BT_ERR("No valid network");
+			return -EINVAL;
+		}
+
+		comp = bt_mesh_comp_get();
+		if (comp == NULL) {
+			BT_ERR("comp == NULL");
+			return -EINVAL;
+		}
+
+		subnet = bt_mesh_mgmt_subnet_get(net_idx);
+		if (subnet == NULL) {
+			BT_ERR("subnet == NULL");
+			return -EINVAL;
+		}
+
+		node = bt_mesh_mgmt_node_alloc(addr, comp->elem_count, net_idx);
+		if (node == NULL) {
+			BT_ERR("node == NULL");
+			return -EINVAL;
+		}
+
+		addr = node->addr;
```

Comment:
Will this ever change the address? We need to do this before we can start provisioning other devices anyway, right?

Reply:
Hum yea I guess that's usually the case. Wasn't there a discussion about the provisioner not having to be part of the network?